src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,60 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+type Props = {
+  setStatus: (status: () => string) => void;
```

Comment:
The `setStatus` function should accept a string directly, not a function returning a string. Change the type to `(status: string) => void;`.